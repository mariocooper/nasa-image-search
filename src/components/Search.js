import React, { useState } from "react";
import "../styles/search.css"
import getImages from "../requests/getImages";

function Search() {
    const [value, setValue] = useState();

    function handleSubmit(event) {
        event.preventDefault();
        getImages(value);
    };

    return (
        <div className="search">
        <form className="search-form" onSubmit={handleSubmit}>
        <input  className="search-input" type="text" onChange={(e) => setValue(e.target.value)}></input>
        <button className="search-button" type="submit">Search</button>
        </form>
        </div>
    )
}
export default Search;
