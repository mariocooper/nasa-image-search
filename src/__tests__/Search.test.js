import React from "react";
import { render, screen } from "@testing-library/react";
import Search from "../components/Search";

describe("Search", () => {
    const { asFragment } = render(<Search />);

    it("renders correctly", () => {
        expect(asFragment()).toMatchSnapshot();
        });

    it("renders a search box", () => {
        render(< Search />);
        expect(screen.getByRole("textbox")).toBeInTheDocument();
        });
});
