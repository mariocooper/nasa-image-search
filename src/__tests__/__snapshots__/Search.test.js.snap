// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search renders correctly 1`] = `
<DocumentFragment>
  <input
    class="search-input"
    type="text"
  />
  <form
    class="search-form"
  />
  <button
    class="search-button"
    type="submit"
  >
    Search
  </button>
</DocumentFragment>
`;
